long-ride-alerts/src/test/java/org/apache/flink/training/exercises/longrides/LongRidesUnitTest.java

```diff
@@ -0,0 +1,119 @@
+package org.apache.flink.training.exercises.longrides;
+
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.streaming.api.operators.KeyedProcessOperator;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.util.KeyedOneInputStreamOperatorTestHarness;
+import org.apache.flink.training.exercises.common.datatypes.TaxiRide;
+import org.apache.flink.training.exercises.testing.ComposedKeyedProcessFunction;
+import org.apache.flink.training.solutions.longrides.LongRidesSolution;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+public class LongRidesUnitTest extends LongRidesTestBase {
+
+    private KeyedOneInputStreamOperatorTestHarness<Long, TaxiRide, Long> harness;
+
+    private KeyedProcessFunction<Long, TaxiRide, Long> javaExercise =
+            new LongRidesExercise.AlertFunction();
+
+    private KeyedProcessFunction<Long, TaxiRide, Long> javaSolution =
+            new LongRidesSolution.AlertFunction();
+
+    protected ComposedKeyedProcessFunction composedAlertFunction() {
+        return new ComposedKeyedProcessFunction(javaExercise, javaSolution);
+    }
```

Comment:
FYI: Scala tests prevent us from using a proper generic return type here
```suggestion
    protected ComposedKeyedProcessFunction composedAlertFunction() {
        return new ComposedKeyedProcessFunction<>(javaExercise, javaSolution);
    }
```